source/user/dft.rst

```diff
@@ -6,495 +6,145 @@ Density functional theory (DFT)
+
+* One need in general be careful with the Libxc convention of GGA functionals, in which the LDA contribution is included
+
+For completeness, it's worth mentioning that yet another way to customize *xc* functionals exists, which uses the :py:meth:`eval_xc` method of the numerical integral class:
+
+  >>> def eval_xc(xc_code, rho, spin=0, relativity=0, deriv=1, verbose=None):
+  >>>     # A fictitious functional to demonstrate the usage
+  >>>     rho0, dx, dy, dz = rho
+  >>>     gamma = (dx ** 2 + dy ** 2 + dz ** 2)
+  >>>     exc = .01 * rho0 ** 2 + .02 * (gamma + .001) ** .5
+  >>>     vrho = .01 * 2 * rho0
+  >>>     vgamma = .02 * .5 * (gamma + .001) ** (-.5)
+  >>>     vlapl = None
+  >>>     vtau = None
+  >>>     vxc = (vrho, vgamma, vlapl, vtau)
+  >>>     fxc = None # 2nd-order functional derivative
+  >>>     kxc = None # 3rd-order functional derivative
+  >>> return exc, vxc, fxc, kxc
+  >>> dft.libxc.define_xc_(mf_hf._numint, eval_xc, xctype='GGA')
+  >>> mf_hf.kernel()
+
+By calling the :func:`dft.libxc.define_xc_` function, the customized :func:`eval_xc` function is patched to the numerical integration class ``DFT._numint`` dynamically.
+
+For more examples of DFT *xc* functional customization, cf. `dft/24-custom_xc_functional.py <https://github.com/pyscf/pyscf/blob/master/examples/dft/24-custom_xc_functional.py>`_ and `dft/24-define_xc_functional.py <https://github.com/pyscf/pyscf/blob/master/examples/dft/24-define_xc_functional.py>`_.
+
+.. _user_dft_numint:
 
 Numerical integration grids
 ===========================
-PySCF implements several numerical integration grids,
-which can be tuned in DFT calculations following the examples in 
-:source:`examples/dft/11-grid_scheme.py`.
-In addition, these grids can be used for the general numerical evaluation of
-basis functions, electron densities, and integrals.
-Some examples can be found in 
-:source:`examples/dft/30-ao_value_on_grid.py`, and
-:source:`examples/dft/31-xc_value_on_grid.py`.
-The following is an example that computes the kinetic energy from the 
-nonnegative kinetic energy density
+
+PySCF implements several numerical integration grids, which can be tuned in KS-DFT calculations following the examples in `dft/11-grid_scheme.py <https://github.com/pyscf/pyscf/blob/master/examples/dft/11-grid_scheme.py>`_. For instance, predefined grid may be set by using levels from ``0`` (very sparse) to ``9`` (very dense), with a default values of ``3``, cf. `pyscf/dft/gen_grid.py <https://github.com/pyscf/pyscf/blob/master/pyscf/dft/gen_grid.py>`_ for more details. Likewise, the default integration grids use Bragg radii for atoms, Treutler-Ahlrichs radial grids, Becke partitioning for grid weights, the pruning scheme of NWChem, and mesh grids, which are all setting that may be overwritten:
```

Comment:
I guess these grid schemes are from Turbomole, originally?

Reply:
I trust your knowledge on this one :). Please see commit e451415

Reply:
OK